crates/derive/src/online/alloy_providers.rs

```diff
@@ -59,6 +59,19 @@ impl AlloyChainProvider {
         Self::new(inner)
     }
 
+    /// Returns the latest L2 block number.
+    pub async fn latest_block_number(&mut self) -> Result<u64> {
+        let b: TransportResult<alloc::string::String> =
+            self.inner.raw_request("eth_blockNumber".into(), ()).await;
```

Comment:
I believe there's a function for this on the provider itself, yeah? [`Provider::get_block_number`](https://docs.rs/alloy-provider/latest/alloy_provider/trait.Provider.html#method.get_block_number)